example_pimpl.cpp

```diff
@@ -0,0 +1,42 @@
+#include "example_pimpl.h"
+#include <string>
+#include <cassert>
+
+class pimpl {
+public:
+    const std::string& get_name() const noexcept { return name_; }
+    void set_name(const std::string& name) { name_ = name; }
+
+private:
+    std::string name_;
+};
+
+example_pimpl::example_pimpl()
+:    pimpl_(new pimpl)
+{
+}
+
+example_pimpl::example_pimpl(char const * const name)
+:    pimpl_(new pimpl)
+{
+    assert(name);
+    pimpl_->set_name(name);
+}
+
+char const * const example_pimpl::get_name() const noexcept
+{
+    assert(!pimpl_->get_name().empty());
+    return pimpl_->get_name().c_str();
+}
+
+void example_pimpl::set_name(char const * const name)
+{
+    assert(name);
+    pimpl_->set_name(name);
+}
+
```

Comment:
Note the below line force instantiation of the rule of zero functions in the translation unit where the pimpl inner class is defined.  This issue already exists for pimpl using std::unique_ptr, though in this case, you would only need to do so for the destructor.  Details on this in a Herb Sutter article here: https://herbsutter.com/gotw/_101/